### PR TITLE
Media Library: State restoration

### PR DESCRIPTION
**Pull request by frosty**
This PR adds state restoration to the media library. Both the main media library and the individual item view controller should be restored. We encode the blog for the media library, and the viewed media item for the item VC.

The PR looks a little bigger than you might expect just because I pulled the media image table view row and immutable row out into their own file to clean things up a little – but none of the content has changed in there.

To test:

* You know the dance. Open one of the media screens in the app, background the app, stop it in Xcode. Reopen and check the correct screen is restored.

Needs review: @kurzee

### CODE REVIEW

**Comment by kurzee** [comment]
Would be a bonus if the `MediaLibraryViewController` restored the scroll position, however I'd imagine this to be tricky. Maybe for another PR if it's doable.

**Comment by kurzee** [comment]
When restoring a selected Media item that has been deleted remotely, the item restores, but trying to interact with the Media object will crash, such as editing the detail info.

This could probably be handled in a separate PR, as I imagine there might be some delicate handling there.

**Comment by frosty** [comment]
So I added in some fixes for manipulating a deleted item after restoration, but then I discovered that there were issues with the collection view when it was restored below the `MediaItemViewController`. This radar summarises the issue fairly well: http://www.openradar.me/15262692. 

I spent some time coding up the workaround discussed in that radar (essentially, accessing the collection view's item count in `controllerWillChangeContent`). It stopped the error from being logged on restoration, but still left the collection view in an inconsistent state. It also caused the original error to surface when just restoring to the media library VC, not the item VC.

TLDR; collection view is buggy, so I removed restoration from `MediaItemViewController`. Whilst it's a shame not to have restoration in there, it does fix the problem and I've probably spent long enough on this already!

@kurzee Care to test it one more time?

File: WordPress/Classes/ViewRelated/Media/MediaItemViewController.swift [hunk 1]
```diff
@@ -79,10 +79,26 @@ class MediaItemViewController: UITableViewController {
...
+
         updateViewModel()
         tableView.reloadData()
     }
 
+    private var isMediaDeleted: Bool {
+        return media.isDeleted || media.managedObjectContext == nil
+    }
+
+    private func handleDeletedMedia() {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Media/MediaItemViewController.swift]: I think it's probably okay to leave it as-is? The view controller gets popped off straight away anyways, and media is currently a constant.
